sdcm/microbenchmarking.py

```diff
@@ -0,0 +1,158 @@
+        results = tests_filtered["hits"]["hits"]
+        sorted_by_type = defaultdict(list)
+        for res in results:
+            if res["_source"]["hostname"] != HOSTNAME:
+                continue
+            sorted_by_type[res["_type"]].append(res)
+
+        report_results = defaultdict(dict)
+        # report_results = {
+        #     "large-partition-skips_1-0.1": {
+        #           "aio":{
+        #               "min"
+        #               "max":
+        #               "cur":
+        #               "last":
+        #               "diff":
+        #           }
+        # }
+        allowed_stats = ("Current", "Last", "Diff last [%]", "Best", "Diff best [%]")
+
+        def set_results_by_param(param):
+            list_of_results_from_db.sort(key=lambda x: datetime.datetime.strptime(x["_id"], "%Y-%m-%d_%H:%M:%S"))
+
+            def get_param_val(x):
+                return float(x["_source"]["results"]["stats"][param])
+
+            def get_commit_id(x):
+                return x["_source"]['versions']['scylla-server']['commit_id']
+
+            last_val = get_param_val(list_of_results_from_db[-1])
+            last_commit = get_commit_id(list_of_results_from_db[-1])
+            cur_val = float(current_result["results"]["stats"][param])
+            min_result = min(list_of_results_from_db, key=get_param_val)
+            min_result_val = get_param_val(min_result)
+            min_result_commit = get_commit_id(min_result)
+            diff_last = ((cur_val - last_val)/max(cur_val, last_val))*100 if max(cur_val, last_val) else 0
+            diff_best = ((cur_val - min_result_val)/max(cur_val, min_result_val))*100 if max(cur_val, min_result_val) else 0
+            stats = {
+                "Current": cur_val,
+                "Last": (last_val, last_commit),
+                "Best": (min_result_val, min_result_commit),
+                "Diff last [%]": diff_last,  # diff in percents
+                "Diff best [%]": diff_best,
+            }
+            report_results[test_type][param] = stats
+
+        for test_type, current_result in current_results.iteritems():
+            list_of_results_from_db = sorted_by_type[test_type]
+            set_results_by_param("aio")
+            set_results_by_param("frag/s")
+            set_results_by_param("cpu")
+            set_results_by_param("time (s)")
+
+        version_info = current_results[current_results.keys()[0]]['versions']['scylla-server']
+        subject = "Microbenchmarks - Performance Regression"
+        for_render = {
+            "subject": subject,
+            "results": report_results,
+            "stats_names": allowed_stats,
+            "kibana_url": self._conf.get('kibana_url'),
+        }
+        for_render.update(dict(test_version=version_info))
+        html = self.render_to_html(for_render, save_to_file="/tmp/perf.html")
+        self.send_email(subject, html)
+
+
+def get_results(results_path, update_db):
```

Comment:
if this function will be called by test, why not to pass index and test_id from the test? The timestamp here is test end time actually.